lib/publiccloud/provider.pm

```diff
@@ -472,9 +472,11 @@ sub terraform_apply {
         }
         if (is_ec2) {
             my $vpc_security_group_ids = script_output("aws ec2 describe-security-groups --region '" . $self->provider_client->region . "' --filters 'Name=group-name,Values=tf-sg' --query 'SecurityGroups[0].GroupId' --output text");
-            my $subnet_id = script_output("aws ec2 describe-subnets --region '" . $self->provider_client->region . "' --filters 'Name=tag:Name,Values=tf-subnet' --query 'Subnets[0].SubnetId' --output text");
+            my $availability_zone = script_output("aws ec2 describe-instance-type-offerings --location-type availability-zone  --filters Name=instance-type,Values=m6g.medium  --region '" . $self->provider_client->region . "' --query 'InstanceTypeOfferings[0].Location' --output 'text'");
+            my $subnet_id = script_output("aws ec2 describe-subnets --region '" . $self->provider_client->region . "' --filters 'Name=tag:Name,Values=tf-subnet' 'Name=availabilityZone,Values=" . $availability_zone . "' --query 'Subnets[0].SubnetId' --output text");
             my $ipv6_address_count = get_var('PUBLIC_CLOUD_EC2_IPV6_ADDRESS_COUNT', 1);
             $cmd .= "-var 'vpc_security_group_ids=$vpc_security_group_ids' " if ($vpc_security_group_ids);
+            $cmd .= "-var 'availability_zone=$availability_zone' " if ($availability_zone);
```

Comment:
initial idea of conditional add of variable was based on optional nature of some variables , e.g. you may run job where you `"-var 'enable_confidential_vm=true' "` is defined but you may also run it without this .  I think in this case `if` is not appropriate . Also I would remove it for `subnet_id` 
```suggestion
            $cmd .= "-var 'availability_zone=$availability_zone' "
```

Reply:
I also did the same for the VPC.